File: assisted-inject-test-java/src/test/java/com/example/assisted/Greeting.java
package com.example.assisted;

import io.micronaut.context.annotation.Parameter;
import io.micronaut.context.annotation.Prototype;

/**
 * A simple product class that combines an injected service with a runtime parameter.
 */
@Prototype
public class Greeting {
  private final GreetingService service;
  private final String name;

  public Greeting(GreetingService service, @Parameter String name) {
    this.service = service;
    this.name = name;
  }

  public String greet() {
    return service.format(name);
  }
}

File: assisted-inject-test-ksp/src/test/kotlin/com/example/assisted/Greeting.kt
package com.example.assisted

import io.micronaut.context.annotation.Parameter
import io.micronaut.context.annotation.Prototype

/**
 * A simple product class that combines an injected service with a runtime parameter.
 */
@Prototype
class Greeting(private val service: GreetingService, @param:Parameter private val name: String) {
  fun greet(): String = service.format(name)
}
